torch/lib/c10d/ProcessGroupRoundRobin.cpp

```diff
@@ -0,0 +1,102 @@
+#include <c10d/ProcessGroupRoundRobin.hpp>
+
+namespace c10d {
+
+ProcessGroupRoundRobin::ProcessGroupRoundRobin(
+    std::vector<std::shared_ptr<ProcessGroup>> processGroups)
```

Comment:
Add checks to enforce same rank and size?

Reply:
Added checks.